### PR TITLE
Fix C++ warnings

### PR DESCRIPTION
**Pull request by AtariDreams**

### CODE REVIEW

File: include/vcpkg/base/cofffilereader.h [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 #include <vcpkg/base/files.h>
 
-#include <stdint.h>
+#include <cstdint>
```
Comment (BillyONeal) [inline include/vcpkg/base/cofffilereader.h:5]: This change is not correct: we refer to the `::global` names, not the `std::` ones. Ditto elsewhere.

File: include/vcpkg/base/json.h [hunk 2]
```diff
@@ -287,7 +287,7 @@ namespace vcpkg::Json
                 ++underlying_;
                 return *this;
             }
-            const_iterator operator++(int) noexcept
+            const const_iterator operator++(int) noexcept
```
Comment (BillyONeal) [inline include/vcpkg/base/json.h:290]: There is no reason to return a const prvalue. What generated this warning?

File: include/vcpkg/base/parse.h [hunk 3]
```diff
@@ -126,7 +126,7 @@ namespace vcpkg
         char32_t cur() const { return m_it == m_it.end() ? Unicode::end_of_file : *m_it; }
         SourceLoc cur_loc() const { return {m_it, m_start_of_line, m_row, m_column}; }
         TextRowCol cur_rowcol() const { return {m_row, m_column}; }
-        char32_t next();
+        virtual char32_t next();
```
Comment (BillyONeal) [inline include/vcpkg/base/parse.h:129]: override?

File: include/vcpkg/base/sortedvector.h [hunk 4]
```diff
@@ -17,9 +17,9 @@ namespace vcpkg
 
         SortedVector() : m_data(), m_comp() { }
         SortedVector(const SortedVector&) = default;
-        SortedVector(SortedVector&&) = default;
+        SortedVector(SortedVector&&) noexcept = default;
```
Comment (BillyONeal) [inline include/vcpkg/base/sortedvector.h:20]: =default implies noexcept if it is; this is kind of a scary change.

File: include/vcpkg/versions.h [hunk 5]
```diff
@@ -37,7 +37,7 @@ namespace vcpkg
         Version right;
 
         VersionDiff() noexcept;
-        VersionDiff(const Version& left, const Version& right);
+        VersionDiff(Version  left, Version  right) noexcept;
```
Comment (BillyONeal) [inline include/vcpkg/versions.h:40]: Version contains a string so I think const& was correct.

File: include/vcpkg/versions.h [hunk 6]
```diff
@@ -74,9 +74,9 @@ namespace vcpkg
         std::string port_name;
         Version version;
 
-        VersionSpec(const std::string& port_name, const Version& version);
+        VersionSpec(std::string  port_name, Version  version);
```
Comment (BillyONeal) [inline include/vcpkg/versions.h:77]: Ditto I think const& was correct.

File: src/vcpkg/commands.integrate.cpp [hunk 7]
```diff
@@ -620,7 +620,7 @@ namespace vcpkg::Commands::Integrate
         static const std::string FISH = "x-fish";
     }
 
-    static std::vector<std::string> valid_arguments(const VcpkgPaths&)
+    static std::vector<std::string> valid_arguments(const VcpkgPaths& paths VCPKG_UNUSED)
```
Comment (BillyONeal) [inline src/vcpkg/commands.integrate.cpp:623]: I think the previous form that worked with compilers before `[[maybe_unused]]` was a thing was better. Ditto elsewhere.

**Comment by AtariDreams** [comment]
Ready!

File: include/vcpkg/base/parse.h (hunk 3 above)
Comment (AtariDreams) [reply include/vcpkg/base/parse.h:129]: Does not compile

**Pull request closed without merging**